llvm/lib/CodeGen/GlobalISel/IRTranslator.cpp

```diff
@@ -1562,8 +1562,9 @@ bool IRTranslator::translateBitCast(const User &U,
 
 bool IRTranslator::translateCast(unsigned Opcode, const User &U,
                                  MachineIRBuilder &MIRBuilder) {
-  if (U.getType()->getScalarType()->isBFloatTy() ||
-      U.getOperand(0)->getType()->getScalarType()->isBFloatTy())
+  if (Opcode != TargetOpcode::G_BITCAST &&
```

Comment:
I think bitcast can still be supported even if we don't know how to convert from bfloat to other types mentioned in #71470. If so, we might want a separate patch for this.

Reply:
Unrelated, but this also misses hidden casts implied by the ABI

Reply:
This change should be done separately

Reply:
This is actually an orthogonal problem. Global ISel is completely broken for bf16 and whatever the outcome of the supporting bf16 in codegen is we just need to be ready some gisel tests will fail and will need to be disabled.